src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,105 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int capacity;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        capacity = DEFAULT_CAPACITY;
+        threshold = (int) (LOAD_FACTOR * capacity);
+        table = (Node<K, V>[]) new Node[capacity];
+    }
+
+    private void resize() {
+        capacity = capacity << 1;
+        threshold = (int) (capacity * LOAD_FACTOR);
+        Node<K, V>[] localTable = table;
+        table = (Node<K, V>[]) new Node[capacity];
+        for (int i = 0; i < localTable.length; i++) {
+            while (localTable[i] != null) {
+                put(localTable[i].key, localTable[i].value);
+                localTable[i] = localTable[i].next;
+                size--;
+            }
+        }
+    }
+
+    private void setNode(Node<K, V> node) {
+        Node<K, V> localNode = table[node.hash % capacity];
+        if (localNode == null) {
+            table[node.hash % capacity] = node;
+            return;
+        }
+        while (localNode.next != null) {
+            localNode = localNode.next;
+        }
+        localNode.next = node;
+    }
+
+    private boolean checkKeysDuplicate(K key, V value) {
```

Comment:
can we combine this method with setNode() method and remove one while loop?

Reply:
done